vcenter_operator/vcenter_sso.py

```diff
@@ -0,0 +1,205 @@
+import logging
+import time
+
+import pyVmomi
+from pyVim import sso
+from pyvmomi_extended import SSO_VERSION
+
+LOG = logging.getLogger(__name__)
+
+
+class SSOSkippedError(Exception):
+    """Exception to skip SSO connection attempts"""
+    pass
+
+
+class VCenterSSO:
+    def __init__(self, dry_run=False):
+        self.dry_run = dry_run
+        self.saml_token = None
+        self.sso_admin_instances = dict()
+        self.ad_ttu_username = None
+        self.ad_ttu_password = None
+        self.domain = "vsphere.local"
+
+    def set_ad_ttu_credentials(self, username, password):
+        """Set the credentials for the AD TTU user"""
+        self.ad_ttu_username = username
+        self.ad_ttu_password = password
```

Comment:
Why does this not reset all our `self.sso_admin_instances()`? Would it maybe make sense to create a new client when we get an update to our credentials?